Runtime/Scripts/Room.cs

```diff
@@ -338,13 +340,33 @@ internal void OnEventReceived(RoomEvent e)
                         ConnectionQualityChanged?.Invoke(quality, participant);
                     }
                     break;
-                case RoomEvent.MessageOneofCase.DataReceived:
+                case RoomEvent.MessageOneofCase.DataPacketReceived:
                     {
-                        var dataInfo = e.DataReceived;
-                        var data = new byte[dataInfo.Data.Data.DataLen];
-                        Marshal.Copy((IntPtr)dataInfo.Data.Data.DataPtr, data, 0, data.Length);
-                        var participant = GetParticipant(e.DataReceived.ParticipantSid);
-                        DataReceived?.Invoke(data, participant, dataInfo.Kind, dataInfo.Topic);
+                        var kind = e.DataPacketReceived.ValueCase;
+                        switch(kind)
+                        {
+                            case DataPacketReceived.ValueOneofCase.None:
+                                //do nothing.
+                                break;
+                            case DataPacketReceived.ValueOneofCase.User:
+                                {
+                                    var dataInfo = e.DataPacketReceived.User;
+                                    var data = new byte[dataInfo.Data.Data.DataLen];
+                                    Marshal.Copy((IntPtr)dataInfo.Data.Data.DataPtr, data, 0, data.Length);
+                                    var participant = GetParticipant(e.DataPacketReceived.ParticipantSid);
```

Comment:
I think we need to use identities here? we will deprecate SID from data messages eventually

Reply:
Yes, I am going to change all `ParticipantSid` to `ParticipantIdentity` for all other places in another PR